crates/common/consensus/src/misc.rs

```diff
@@ -0,0 +1,170 @@
+use std::cmp::max;
+
+use alloy_primitives::{aliases::B32, B256};
+use ethereum_hashing::{hash, Context, Sha256Context};
+use tree_hash::{Hash256, TreeHash};
+
+use crate::{
+    deneb::beacon_state::BeaconState,
+    fork_choice::helpers::constants::{
+        MAX_EFFECTIVE_BALANCE, MAX_SEED_LOOKAHEAD, SHUFFLE_ROUND_COUNT, SLOTS_PER_EPOCH,
+    },
+    fork_data::ForkData,
+};
+
+static MAX_RANDOM_BYTE: u64 = 255;
+
+pub fn compute_shuffled_index(index: usize, index_count: usize, seed: B256) -> usize {
+    assert!(index < index_count);
+
+    let mut index = index;
+    for round in 0..SHUFFLE_ROUND_COUNT {
+        let pivot = bytes_to_int64(&hash_with_round(seed, round)[..]) % index_count as u64;
+        index = do_round(seed, index, pivot as usize, round, index_count);
+    }
+    index
```

Comment:
```suggestion
pub fn compute_shuffled_index(index: usize, index_count: usize, seed: B256) -> anyhow::Result<usize> {
    ensure!(index < index_count, "xyz");

    let mut index = index;
    for round in 0..SHUFFLE_ROUND_COUNT {
        let pivot = bytes_to_int64(&hash_with_round(seed, round)[..]) % index_count as u64;
        index = do_round(seed, index, pivot as usize, round, index_count);
    }
    Ok(index)
```

I think it is better to use a result here